### ISSUE TITLE
Query diagnostics

### ISSUE THREAD

**Issue by pveentjer** [hazelcast/hazelcast#11234]

**Comment by mmedenjak** [comment]
@taburet can we close this issue? Do we have some kind of query statistics with the new index diagnostics?

**Comment by taburet** [comment]
For now we have really basic query statistics: number of queries executed and number of indexed queries executed on a certain map, the rest is per-index statistics. In other words, we don't know the exact queries and their performance. Seems like Peter is thinking about per-query stats collection on which slow query detection and query plan optimizer may be based.

**Comment by mmedenjak** [comment]
Also related: https://github.com/hazelcast/hazelcast-enterprise/issues/1924

**Comment by mmedenjak** [comment]
Since we're heavily invested in the new query engine, I don't think we should spend too much time on this. Maybe we should think more about how we will integrate the new query engine with diagnostics cc @devozerov

**Comment by devozerov** [comment]
Yes, I think that we should first focus on monitoring for SQL instead of the old predicate API
